### PR TITLE
Federation require support

### PR DESCRIPTION
**Pull request by kmoore-intuit**
# What Changed

# Why

Todo:

- [x] Add tests
- [ ] Add docs

### CODE REVIEW

File: src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    @VisibleForTesting
+    RequireValidator requireValidator = new RequireValidator();
+
+    @Override
+    public XtextGraph transform(final XtextGraph source) {
+
+        if(source.getServiceProvider().isFederationProvider()) {
+            for(final TypeDefinition typeDefinition : source.getTypes().values()) {
+                List<FieldDefinition> fieldDefinitions = (typeDefinition instanceof ObjectTypeDefinition || typeDefinition instanceof InterfaceTypeDefinition) ?
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java:28]: should this be a util ?

File: src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                        requireValidator.validate(source, typeDefinition, requireDirective);
+                    }
+                }
+            }
+        }
+
+        return source;
+    }
+
+    private Directive getRequireArgument(FieldDefinition fieldDefinition) {
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java:43]: there should already be a util for this as well

Comment (ashpak-shaikh) [reply src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java:43]: pass the name of the directive and fielddefintion and it gives u directive

File: src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+public class RequireTransformer implements Transformer<XtextGraph, XtextGraph> {
+
+    @VisibleForTesting
+    RequireValidator requireValidator = new RequireValidator();
+
+    @Override
+    public XtextGraph transform(final XtextGraph source) {
+
+        if(source.getServiceProvider().isFederationProvider()) {
+            for(final TypeDefinition typeDefinition : source.getTypes().values()) {
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java:27]: try using streams here

Comment (kmoore-intuit) [reply src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java:27]: I was suppose to look at that before I push; thanks for the reminder!

File: src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import com.intuit.graphql.orchestrator.federation.requiresdirective.RequireValidator;
+import com.intuit.graphql.orchestrator.xtext.XtextGraph;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.intuit.graphql.orchestrator.utils.FederationUtils.FEDERATION_REQUIRES_DIRECTIVE;
+import static com.intuit.graphql.orchestrator.utils.XtextTypeUtils.getFieldDefinitions;
+
+public class RequireTransformer implements Transformer<XtextGraph, XtextGraph> {
```
[the remark this answers is not in the document]
Comment (kmoore-intuit) [reply src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java:18]: Yes, it is

File: src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java (hunk 2 above)
Comment (kmoore-intuit) [reply src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java:43]: Did not exist, but I just created it

**Review by CNAChino — APPROVED** [review]

**Pull request merged**